utils/tfrecords/converter.py

```diff
@@ -0,0 +1,200 @@
+        type=str,
+        help='The attribute name based on which the object can identified'
+    )
+
+    return parser.parse_args()
+
+def process_cvat_xml(args):
+  """Transforms a single XML in CVAT format to tfrecords.
+  """
+
+  train_percentage = int(args.train_percentage)
+  assert (train_percentage<=100 and train_percentage>=0)
+
+  cvat_xml = ET.parse(args.cvat_xml).getroot()
+
+  output_dir = Path(args.output_dir)
+  if not output_dir.exists():
+    print("Creating the output directory because it doesn't exist")
+    output_dir.mkdir()
+
+  cvat_name, output_dir, min_train = \
+          args.attribute, output_dir.absolute(), args.min_train
+
+  # Open the tfrecord files for writing
+  writer_train = tf.python_io.TFRecordWriter(
+      os.path.join(output_dir.absolute(), 'train.tfrecord'))
+  writer_eval  = tf.python_io.TFRecordWriter(
+      os.path.join(output_dir.absolute(), 'eval.tfrecord'))
+
+  # extract the object names
+  object_names = []
+  num_imgs = 0
+  for img in cvat_xml.findall('image'):
+        num_imgs += 1
+        for box in img:
+            if cvat_name == "" :
+                object_names.append(box.attrib['label'])
+            else :
+                for attribute in box :
+                    if attribute.attrib['name'] == cvat_name :
+                        object_names.append(attribute.text.lower())
+
+  labels, values = zip(*Counter(object_names).items())
+
+  # Create the label map file
+  saved_dict = dict()
+  reverse_dict = dict()
+  with codecs.open(os.path.join(output_dir,'label_map.pbtxt'),
+            'w', encoding='utf8') as f:
+        counter = 1
+        for iii, label in enumerate(labels):
+            if values[iii] < min_train :
+                continue
+            saved_dict[label] = counter
+            reverse_dict[counter] = label
+            f.write(u'item {\n')
+            f.write(u'\tid: {}\n'.format(counter))
+            f.write(u"\tname: '{}'\n".format(label))
+            f.write(u'}\n\n')
+            counter+=1
+
+  num_iter = num_imgs
+  eval_num = num_iter * (100 - train_percentage) // 100
+  train_num = num_iter - eval_num
+
+
+  for counter,example in enumerate(cvat_xml.findall('image')):
+    tf_example = create_tf_example(example, args.attribute, saved_dict,  args.image_dir)
+    if(counter < train_num):
+        writer_train.write(tf_example.SerializeToString())
+    else :
+        writer_eval.write(tf_example.SerializeToString())
+
+  writer_train.close()
+  writer_eval.close()
+
+
+  return saved_dict, num_imgs
+
+
+# Defining the main conversion function
+def create_tf_example(example, cvat_name, saved_dict, img_dir):
+  # Process one image data per run
+  height = int(example.attrib['height']) # Image height
+  width = int(example.attrib['width']) # Image width
+  filename = os.path.join(img_dir, example.attrib['name'])
+  filename = filename.encode('utf8')
+  with tf.gfile.GFile(filename,'rb') as fid:
+    encoded_jpg = fid.read()
```

Comment:
What if the image directory contains png or bmp or mix of files?

Reply:
Following this guide [USING YOUR OWN DATASET](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md), it seems like the `dataset_utils` does only support `.jpg` and `.png`, I will process these two cases, show a warning and leave if it does find some other image extension.